### ISSUE TITLE
iOS Tailscale client not connecting to Reolink camera

### ISSUE THREAD

**Issue by gauravpandeydotdev** [tailscale/tailscale#9423]
### What is the issue?

I set up a **Reolink** camera(E1 Zoom) recently.

I assigned it a **static IP** `10.0.1.XX`.

I have tailscale clients on my:
1) Synology (exit node, subnet advertising)
2) iPhone
3) iPad
4) Mac Studio

Then on my Synology tailscale nodes, I ran: `tailscale up --advertise-routes=10.0.1.XX/32,10.0.1.YY/32 --advertise-exit-node`

Now, when I connect my iPad and Mac Studio to a non-local area network (cellular hotspot or another Wi-Fi network) and enable Tailscale without enabling the exit node, I can access my Reolink camera using the Reolink client(MacOS) and the Reolink app(iPad). ✅ 

Now, when I do the same with my iPhone, I can not access the camera. 🔴  
But when I _enable the exit node_, I can. ✅ 

I can ping the IP of the camera from all devices with and without enabling the exit node, but I am confused as to why on iPhone, I have to enable the exit node to access the camera. 

**Hunch**: The iOS client of Tailscale only allows TCP and not UDP, but when you enable the exit node, then all calls are routed to the exit node and hence it works. But then why is this not happening in iPad? I would have assumed that iPad would be a fork of iOS.

### Steps to reproduce

1) set up a Reolink E1 Zoom camera
2) switch your iPhone to a non-local area network
3) turn on tailscale VPN
4) try to access camera using the reolink app
5) enable exit node
6) try to access camera using the reolink app again

### OS

iOS

### OS version

16.6.1

### Tailscale version

1.48.2

**Comment by afnash1998** [comment]
I was able to reproduce this issue with a Reolink Duo 2 camera and my iPhone. I can access the camera dashboard via the camera's IP address and through Tailscale, but it does not work with the iOS app. When I use the subnet router as an exit node, the iOS app works.

**Comment by bradfitz** [comment]
Are you accepting subnet routes on the iPhone? The latest versions on iOS have more settings.

/cc @agottardo @barnstar

**Comment by agottardo** [comment]
@bradfitz Use Tailscale Subnets is enabled on iOS by default, there is no setting for it.

**Comment by Cyphs** [comment]
Any solution besides routing all traffic through an exit node?

**Comment by agottardo** [comment]
@Cyphs your question is better directed to the developers of the app; it sounds like their iPad version is doing something different than the iPhone one. Tailscale on iPhone and iPad is exactly the same (we ship the same app for both platforms).

**Comment by Cyphs** [comment]
> @Cyphs your question is better directed to the developers of the app; it sounds like their iPad version is doing something different than the iPhone one. Tailscale on iPhone and iPad is exactly the same (we ship the same app for both platforms).

Well actually it's not just the Reolink app but also the ASUS Router app, Termius app, etc. almost everything that uses an app for local network access rather than webpages in Safari, and I believe subnets are set up correctly since it was tested to be working as expected in macOS for example.

**Comment by agottardo** [comment]
> Well actually it's not just the Reolink app but also the ASUS Router app, Termius app, etc. almost everything that uses an app for local network access rather than webpages in Safari, and I believe subnets are set up correctly since it was tested to be working as expected in macOS for example.

That is a whole different story then, and is most likely caused by the fact that multicast traffic (often used for device advertisements) is not routed through Tailscale.

**Comment by aaraney** [comment]
@darthtechworker, @Cyphs I was having the same issue. A [reddit user pointed out](https://www.reddit.com/r/reolinkcam/comments/luzkdi/cannot_access_reolink_app_with_vpn_enabled/) the IOS app was failing to connect to the camera because the camera was setup using the QR code on the camera which uses Reolink's UID server, not by IP. After deleting the device in the Reolink app and re-adding it by using the "LAN" option in the add menu, I can now view the camera when connected to tailscale.

**Comment by Cyphs** [comment]
> [@darthtechworker](https://github.com/darthtechworker), [@Cyphs](https://github.com/Cyphs) I was having the same issue. A [reddit user pointed out](https://www.reddit.com/r/reolinkcam/comments/luzkdi/cannot_access_reolink_app_with_vpn_enabled/) the IOS app was failing to connect to the camera because the camera was setup using the QR code on the camera which uses Reolink's UID server, not by IP. After deleting the device in the Reolink app and re-adding it by using the "LAN" option in the add menu, I can now view the camera when connected to tailscale.

Yes it needs to be with the LAN option, but that’s not quite the issue we were discussing. The issue is that it doesn’t work unless you route all traffic through the exit node.

**Comment by aaraney** [comment]
@Cyphs, sorry I should have read more carefully! I can connect to my camera's stream without an exit node selected using the steps I described above.

**Comment by Cyphs** [comment]
> [@Cyphs](https://github.com/Cyphs), sorry I should have read more carefully! I can connect to my camera's stream without an exit node selected using the steps I described above.

Just to confirm, are you on iOS or iPadOS? Sadly on iOS, I’ve still been unable to get it to work without using the exit node, but I believe iPadOS isn’t affected. I’m not using the latest iOS version, however, so perhaps I will have to make soon just in case anything changed since iOS 18.1.

**Comment by aaraney** [comment]
@Cyphs, IOS 18.2!

**Comment by Cyphs** [comment]
> [@Cyphs](https://github.com/Cyphs), IOS 18.2!

One more question, have you tested this on cellular data, in case you’ve only tried Wi-Fi?

**Comment by aaraney** [comment]
Yes, all my tests have been conducted over a cellular network.

**Comment by hsshah** [comment]
I am currently facing the exact issue.  Bizarrely, it used to work earlier and just recently stopped working!  Not sure what changed.

**Comment by Cyphs** [comment]
I still haven’t changed anything because I believe everything should be configured correctly.

However, I’ve discovered something quite interesting with Termius (SSH app)

Again, completely on cellular data and Wi-Fi disabled, I use Tailscale to try to access my local network, devices blocked from Internet in my router:

When trying to SSH into a local device with the exit node disabled, it cannot connect.

When I enable the exit node, it can connect.

I tried to disable the exit node with the SSH session still connected and it still works! Termius can keep active SSH sessions open in the background. I can even toggle the exit node off and on and it doesn’t affect the connection.

Once I quit the SSH session, I cannot connect again to the device in Termius until I enable the exit node again.

I can’t test this behavior with other apps like Reolink because it refreshes the connection once you leave its app, so that didn’t work for it.

I am wondering how this is possible.

And again, local things work fine in Safari with the exit node disabled on cellular data. It’s just apps that are like this.

**Comment by zubieta** [comment]
I'm facing the exact same issue on an iPhone. I just wanted to add that I can go to the web page of Reolink camera from the browser by directly typing the IP address while connected to Tailscale through mobile connection. Even though the app is not able to reach it.

In short, it looks like the browser's connection (Chrome in my case) is going through the VPN but not the app's.

If there is a way to debug this, please let me know, and I'll gladly try to gather any necessary logs.

**Comment by Cyphs** [comment]
One of the only apps that I found so far which works without enabling the Exit Node is Rustdesk. I can simply connect to the remote computer with the LAN IP.

Still no clue why apps like Reolink, Termius (at first; see my above comment), ASUS Router do not connect without using the Exit Node.

**Comment by Cyphs** [comment]
A possible culprit I’ve found is that it seems even if I disable “Limit IP Address Tracking” for the cellular network, and specify custom DNS in the Tailscale app’s DNS settings, it does not actually use it when I test for IP leaks on websites and still use’s my carrier’s DNS. Even after deleting the VPN configuration and adding it again.

It only takes effect when using the exit node.

**Comment by AngusF22** [comment]
Has there been any progress with this? I've had the same issue with multiple iOS devices accessing my Reolink NVR. Using an exit node doesn't help - I have to fully disable Tailscale to access anything Reolink from outside my network.

Thank you!

**Comment by afnash1998** [comment]
I have a Reolink NVR now. I have to toggle the exit node on and wait an additional 15-20 seconds for it to connect. It refuses to work without the exit node enabled. Like before, I can access the Reolink NVR over IP address without enabling the exit node.

**Comment by DjiB1** [comment]
> [@darthtechworker](https://github.com/darthtechworker), [@Cyphs](https://github.com/Cyphs) I was having the same issue. A [reddit user pointed out](https://www.reddit.com/r/reolinkcam/comments/luzkdi/cannot_access_reolink_app_with_vpn_enabled/) the IOS app was failing to connect to the camera because the camera was setup using the QR code on the camera which uses Reolink's UID server, not by IP. After deleting the device in the Reolink app and re-adding it by using the "LAN" option in the add menu, I can now view the camera when connected to tailscale.

thank you so much. I've disabled UID from the web UI and removed the camera from the app. added again via LAN option and tailscale now works, no need for exit node, just advertising the local subnet (which is default). now using the app to vie the live feed is blazing fast local and tailscale!! with Internet disabled from the router. super happy
